### PR TITLE
KOGITO-9098 - Review process events to log actions according to publishing state

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#2483` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT= 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml' -o 'bc' -p kiegroup/kogito-apps -u https://github.com/kiegroup/kogito-runtimes/pull/2961 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-apps/2483/display/redirect

**Test results:**
- PASSED: 2153
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-apps/2483/testReport/org.kie.kogito.it.jobs/CallbackStateTimeoutsIT/Build_projects___callbackStateTimeoutsExceeded/">Build projects / org.kie.kogito.it.jobs.CallbackStateTimeoutsIT.callbackStateTimeoutsExceeded</a></summary>
java.lang.RuntimeException: java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details<br/>Caused by: java.lang.IllegalStateException: Unable to determine the status of the running process. See the above logs for details<br/>
</details>

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-apps tests

**Review by fjtirado — COMMENTED** [review]
Hmmm, I do not think to be a good idea, because it will make event publishing synchronous. 
If we want feedback about when the event has been acknwoleged, we should register a callback

File: quarkus/addons/events/process/runtime/src/main/java/org/kie/kogito/events/process/ReactiveMessagingEventPublisher.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ protected void publishToTopic(DataEvent<?> event, Emitter<String> emitter, Strin
             String eventString = json.writeValueAsString(event);
             logger.debug("Event payload '{}'", eventString);
 
-            emitter.send(eventString);
+            emitter.send(eventString).toCompletableFuture().get();
             logger.debug("Successfully published event {} to topic {}", event, topic);
```
Comment (fjtirado) [inline quarkus/addons/events/process/runtime/src/main/java/org/kie/kogito/events/process/ReactiveMessagingEventPublisher.java:110]:
```suggestion
           emitter.send(eventString).thenAccept(e -> logger.debug("Successfully published event {} to topic {}", e, topic) );
```

**Review by fjtirado — APPROVED** [review]

**Comment by cristianonicolai** [comment]
Jenkins rerun kogito-quarkus-examples tests

**Review by nmirasch — APPROVED** [review]

**Pull request merged**